hexl/CMakeLists.txt

```diff
@@ -74,6 +76,17 @@ if(CpuFeatures_FOUND)
     target_include_directories(hexl PUBLIC ${CpuFeatures_INCLUDE_DIR}) # Public headers
 endif()
 
+if (HEXL_FPGA_COMPATIBILITY STREQUAL "1")
```

Comment:
This option only applies for HEXL-FPGA compatibility.  It will be used nowhere else.  Ideally it will be a hidden option, and thus is not documented.  Does HEXL provide a mechanism to add a hidden option?

Reply:
The option is not intended to be documented.  The default behavior keeps the same.

Reply:
I see, well in that case I believe in CMake there is an option to define a variable as an `INTERNAL` [type](https://cmake.org/cmake/help/latest/command/set.html?highlight=internal) that will hide the variable from the user.

Another method which might be preferred is to use [mark_as_advanced](https://cmake.org/cmake/help/latest/command/mark_as_advanced.html) instead so that the variable will only appear when users specifically look at the advanced options.

Reply:
Nice. These two methods will hide an option from GUI. 
I have not used GUI, and didn't know the subtle of these attributes.  Good learning.  Thank you.
Marked the option as "INTERNAL" in commit f5b462a72b971.

Reply:
Ok, looks good. I have not used the GUI either but I have used the TUI on occasion. It's useful to see what variables have been cached if you ever forget.